addons/website_sale_loyalty/controllers/main.py

```diff
@@ -90,8 +93,14 @@ def claim_reward(self, reward, **post):
             if reward_id in rewards:
                 coupon_id = coupon
         redirect = post.get('r', '/shop/cart')
-        if not coupon_id or not reward_id.exists() or reward_id.multi_product:
+        if not coupon_id or not reward_id.exists():
             return request.redirect(redirect)
+        if reward_id.multi_product and "product" in post:
+            product = request.env["product.product"].sudo().browse(int(post["product"]))
+        else:
+            product = None
```

Comment:
Bad practice to put recordsets in the context, only put an id in the context instead.